File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        mapping = {}
        for i, n in enumerate(nums):
            mapping[n] = i

        max_val = float("-Inf")
        for k in mapping.keys():
            if -k in mapping:
                max_val = max(max_val, k)
    
        return max_val if max_val != float("-Inf") else -1
            
        

File: leetcode_solutions/2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int FindMaxK(int[] nums) {
        Dictionary<int, int> mapping = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            mapping[nums[i]] = i;
        }
        int maxVal = int.MinValue;
        foreach (int k in mapping.Keys) {
            if (mapping.ContainsKey(-k)) {
                maxVal = Math.Max(maxVal, k);
            }
        }
        return maxVal != int.MinValue ? maxVal : -1;
    }
}
